core/src/Kokkos_HostSpace.hpp

```diff
@@ -261,12 +261,7 @@ class SharedAllocationRecord<Kokkos::HostSpace, void>
   const Kokkos::HostSpace m_space;
 
  protected:
-  ~SharedAllocationRecord()
-#if defined( \
-    KOKKOS_IMPL_INTEL_WORKAROUND_NOEXCEPT_SPECIFICATION_VIRTUAL_FUNCTION)
-      noexcept
-#endif
-      ;
+  ~SharedAllocationRecord() noexcept;
```

Comment:
Why is it ok to drop the workaround for Intel?

Reply:
Okay, I'll put it back in. It shouldn't actually mean anything, but that's fine; it was probably a workaround for something at some point and someone was worried it would mess up the other builds without realizing that it's implied anyway for destructors. You have to explicitly put `noexcept(false)` to propagate exceptions through a constructor.

Reply:
Fixed in d73f33bc1

Reply:
It was not a loaded question.  I'd be happy to get rid of it